includes/embeds/class-amp-facebook-embed.php

```diff
@@ -47,4 +61,85 @@ public function render( $args ) {
 			)
 		);
 	}
+
+	/**
+	 * Sanitized <div class="fb-video" data-href=> tags to <amp-facebook>
+	 *
+	 * @param DOMDocument $dom DOM.
+	 */
+	public function sanitize_raw_embeds( $dom ) {
+		/**
+		 * Node list.
+		 *
+		 * @var DOMNodeList $node
+		 */
+		$nodes     = $dom->getElementsByTagName( $this->sanitize_tag );
+		$num_nodes = $nodes->length;
+
+		if ( 0 === $num_nodes ) {
+			return;
+		}
+
+		for ( $i = $num_nodes - 1; $i >= 0; $i-- ) {
+			$node = $nodes->item( $i );
+			if ( ! $node instanceof DOMElement ) {
+				continue;
+			}
+
+			if ( $this->is_facebook_raw_embed( $node ) ) {
+				$this->create_amp_facebook_and_replace_node( $dom, $node );
+			}
+		}
+	}
+
+	/**
+	 * Make final modifications to DOMNode
+	 *
+	 * @param DOMNode $node The DOMNode to adjust and replace.
+	 *
+	 * @return boolean
+	 */
+	private function is_facebook_raw_embed( $node ) {
+		$class_attr = $node->getAttribute( 'class' );
+
+		return null !== $class_attr && $node->hasAttribute( 'data-href' )
+			&& ( false !== strpos( $class_attr, 'fb-post' ) || false !== strpos( $class_attr, 'fb-video' ) );
+
+	}
+
+	/**
+	 * Make final modifications to DOMNode
+	 *
+	 * @param DOMDocument $dom The HTML Document.
+	 * @param DOMNode     $node The DOMNode to adjust and replace.
+	 */
+	private function create_amp_facebook_and_replace_node( $dom, $node ) {
+		$embed_type = $this->get_embed_type( $node );
+
+		$amp_facebook_node = AMP_DOM_Utils::create_node( $dom, $this->amp_tag, array(
+			'data-href'     => $node->getAttribute( 'data-href' ),
+			'data-embed-as' => $embed_type,
+			'layout'        => 'responsive',
+			'width'         => $this->DEFAULT_WIDTH,
+			'height'        => $this->DEFAULT_HEIGHT,
+		) );
+
+		$node->parentNode->replaceChild( $amp_facebook_node, $node );
+
+		$this->did_convert_elements = true;
+	}
+
+	/**
+	 * Make final modifications to DOMNode
+	 *
+	 * @param DOMNode $node The DOMNode to adjust and replace.
+	 *
+	 * @return boolean
+	 */
+	private function get_embed_type( $node ) {
```

Comment:
This method could potentially be merged with `is_facebook_raw_embed`, if we don't go the XPath route. The `get_embed_type()` method could also check to see if `data-href` attribute is present, and likewise check if the classes are present.